nextest-runner/src/config.rs

```diff
@@ -283,11 +284,58 @@ struct DefaultProfileImpl {
     failure_output: TestOutputDisplay,
     success_output: TestOutputDisplay,
     fail_fast: bool,
-    #[serde(with = "humantime_serde")]
-    slow_timeout: Duration,
+    #[serde(deserialize_with = "deserialize_slow_timeout")]
+    slow_timeout: SlowTimeout,
     junit: DefaultJunitImpl,
 }
 
+/// Type for the slow-timeout config key.
+#[derive(Clone, Debug, Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub struct SlowTimeout {
+    #[serde(with = "humantime_serde")]
+    pub(crate) period: Duration,
+    pub(crate) shutdown_after: Option<usize>,
```

Comment:
So this should probably be deserialized into an enum:

```
enum KillAfter {
    Never,
    Periods(usize), // can also use NonZeroUsize here
}
```

Then the 0 is deserialized into Never and anything else is deserialized into the `Periods` variant.

Again, the `Option` is a bit confusing here -- if a profile doesn't have the option set, it should fall back to the repository-specific default config, which in turn should fall back to the default profile shipped with nextest. See [Hierarchical configuration](https://nexte.st/book/configuration.html#hierarchical-configuration) for how it should work.

(This is why the separation between `Default` and `Custom` is important -- it is required for hierarchical configuration to work properly. If Rust had higher-kinded types a la Haskell, we could unify the code for these--but unfortunately it doesn't.)

Reply:
i understand, the initial version i proposed did not adhere to the hierarchical configuration logic.
i just pushed a commit that implements it and i'm curious for your feedback. the duplication bothers me and i'd like to avoid it if possible.